src/lib/Simplify.hs

```diff
@@ -183,15 +179,12 @@ simpDeclsSubst
   -> SimplifyM i o (Subst (AtomSubstVal CoreIR) i' o)
 simpDeclsSubst !s = \case
   Empty -> return s
-  Nest (Let b (DeclBinding ann _ expr)) rest -> do
+  Nest (Let b (DeclBinding _ _ expr)) rest -> do
     let hint = (getNameHint b)
-    x <- withSubst s $ case (ann, expr) of
-      (NoInlineLet, Atom a) ->
-        liftM Var $ emitDecl hint NoInlineLet =<< (Atom <$> simplifyAtom a)
-      _ -> simplifyExpr hint expr
```

Comment:
What changed that makes it unnecessary to detect NoInlineLet here?

Reply:
I think it was always dead. `NoInlineLet` is only for top-level decls. I'd like to enforce that syntactically at some point.